src/HW/Main.java

```diff
@@ -1,7 +1,250 @@
+                "Text, \n" +
+                "Date \n" +
+                "FROM Messages \n" +
+                "WHERE Sender = (SELECT ID FROM Users WHERE Nickname = ?) \n" +
+                "OR Receiver = (SELECT ID FROM Users WHERE Nickname = ?) \n" +
+                "OR Receiver = (SELECT ID FROM Users WHERE Nickname = 'admin')");
+    }
+
+    // Получаем nickname по логину и паролю
+    public static String getNicknameByLoginAndPassword(String login, String password) {
+        String nick = null;
+        try {
+            psGetNickname.setString(1, login);
+            psGetNickname.setString(2, password);
+            ResultSet rs = psGetNickname.executeQuery();
+            if (rs.next()) {
+                nick = rs.getString(1);
+            }
+            rs.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return nick;
+    }
 
+    // Проходим регистрацию
+    public static boolean registration(String login, String password, String nickname) {
+        try {
+            psRegistration.setString(1, login);
+            psRegistration.setString(2, password);
+            psRegistration.setString(3, nickname);
+            psRegistration.executeUpdate();
+            return true
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    // Метод смены ника
+    public static boolean changeNick(String oldNick, String newNick) {
+        try {
+            psChangeNick.setString(1, newNick);
+            psChangeNick.setString(2, oldNick);
+            psChangeNick.executeUpdate();
+            return true;
+        } catch (SQLException e) {
+            return false;
+        }
+    }
+
+    // Метод добавления сообщений в БД
+    public static boolean addMessage(String sender, String receiver, String text, String date) {
+        try {
+            psAddMessage.setString(1, sender);
+            psAddMessage.setString(2, receiver);
+            psAddMessage.setString(3, text);
+            psAddMessage.setString(4, date);
+            psAddMessage.executeUpdate();
+            return true;
+        } catch (SQLException e) {
+            return false;
+        }
+    }
+
+    // Метод извлечения сообщений из БД
+    public static String getMessageForNick(String nick) {
+        StringBuilder sb = new StringBuilder();
+        try {
+            psGetMessageForNick.setString(1, nick);
+            psGetMessageForNick.setString(2, nick);
+            ResultSet rs = psGetMessageForNick.executeQuery();
+
+            while (rs.next()) {
+                String sender = rs.getString(1);
+                String receiver = rs.getString(2);
+                String text = rs.getString(3);
+                String date = rs.getString(4);
+                //всем сообщение
+                if (receiver.equals("admin")) {
+                    sb.append(String.format("[ %s ] : %s\n", sender, text));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.